activesupport/lib/active_support/concern.rb

```diff
@@ -123,6 +123,10 @@ def append_features(base)
       end
     end
 
+    # Without an argument, given block is evaluated in context of base class.
```

Comment:
I'm not excited to mention about "With or without an argument".

Ideally we wanted to define `def included(&block)` as only allow block, but Ruby itself already have `Module#included` so `base = nil` is actually a hack for re-using the name `included`.

As similar thing with this, `Relation#count`, `Relation#sum`. Since `Relation` includes `Enumerable`, that relation methods conflicts with `Enumerable#count` and `Enumerable#sum`.

https://github.com/rails/rails/blob/530294270227e4046033609dffd1df08e19ebb24/activerecord/lib/active_record/relation/calculations.rb#L5-L41

https://github.com/rails/rails/blob/530294270227e4046033609dffd1df08e19ebb24/activerecord/lib/active_record/relation/calculations.rb#L79-L84